packages/contracts/src/personal/PersonalSpaceAdminPlugin.sol

```diff
@@ -48,16 +70,193 @@ contract PersonalSpaceAdminPlugin is PluginCloneable, ProposalUpgradeable {
+    function submitAcceptSubspace(
+        IDAO _subspaceDao,
+        address _spacePlugin
+    ) public auth(MEMBER_PERMISSION_ID) {
+        IDAO.Action[] memory _actions = new IDAO.Action[](1);
+        _actions[0].to = _spacePlugin;
+        _actions[0].data = abi.encodeCall(SpacePlugin.acceptSubspace, (address(_subspaceDao)));
+
+        uint256 _proposalId = _createProposal(msg.sender, _actions);
+
+        dao().execute(bytes32(_proposalId), _actions, 0);
+
+        // The event will be emitted by the space plugin
+    }
+
+    /// @notice Creates and executes a proposal that makes the DAO remove the given DAO as a subspace.
+    /// @param _subspaceDao The address of the DAO that holds the subspace to remove
+    /// @param _spacePlugin The address of the space plugin where changes will be executed
+    function submitRemoveSubspace(
+        IDAO _subspaceDao,
+        address _spacePlugin
+    ) public auth(MEMBER_PERMISSION_ID) {
+        IDAO.Action[] memory _actions = new IDAO.Action[](1);
+        _actions[0].to = _spacePlugin;
+        _actions[0].data = abi.encodeCall(SpacePlugin.removeSubspace, (address(_subspaceDao)));
+
+        uint256 _proposalId = _createProposal(msg.sender, _actions);
+
+        dao().execute(bytes32(_proposalId), _actions, 0);
+
+        // The event will be emitted by the space plugin
+    }
+
+    /// @notice Creates and executes a proposal that makes the DAO grant membership permission to the given address
+    /// @param _newMember The address to grant member permission to
+    function submitNewMember(address _newMember) public auth(EDITOR_PERMISSION_ID) {
+        IDAO.Action[] memory _actions = new IDAO.Action[](1);
+        _actions[0].to = address(dao());
+        _actions[0].data = abi.encodeCall(
+            PermissionManager.grant,
+            (address(this), _newMember, MEMBER_PERMISSION_ID)
+        );
+
+        uint256 _proposalId = _createProposal(msg.sender, _actions);
+
+        dao().execute(bytes32(_proposalId), _actions, 0);
+
+        emit MemberAdded(address(dao()), _newMember);
+    }
+
+    /// @notice Creates and executes a proposal that makes the DAO revoke membership permission from the given address
+    /// @param _member The address that will no longer be a member
+    function submitRemoveMember(address _member) public auth(EDITOR_PERMISSION_ID) {
+        IDAO.Action[] memory _actions = new IDAO.Action[](1);
+        _actions[0].to = address(dao());
+        _actions[0].data = abi.encodeCall(
+            PermissionManager.revoke,
+            (address(this), _member, MEMBER_PERMISSION_ID)
+        );
+
+        uint256 _proposalId = _createProposal(msg.sender, _actions);
+
+        dao().execute(bytes32(_proposalId), _actions, 0);
+
+        emit MemberRemoved(address(dao()), _member);
+    }
+
+    /// @notice Creates and executes a proposal that makes the DAO revoke any permission from the sender address
+    function leaveSpace() external {
```

Comment:
We may want to have more granular controls for users leaving a space. e.g., a user can choose whether to leave as just a member, just an editor, or both. Posed this question in discord.

Reply:
See my reply there. This opens the door to confusing edge cases that would need to be resolved on Substream/UI, like that of an Editor who leaves as a member.

Can be done, no problem. But let's confirm your thoughts on the edge cases